leshan-core/src/main/java/org/eclipse/leshan/core/request/argument/Arguments.java

```diff
@@ -0,0 +1,314 @@
+     * DIGIT = "0" / "1" / "2" / "3" / "4" / "5" / "6" / "7" / "8" / "9"
+     * CHAR = "!" / %x23-26 / %x28-5B / %x5D-7E
+     * }
+     * </pre>
+     * 
+     * @param arglist String contains arguments to parse.
+     * @return {@link Arguments} represents list of arguments.
+     * @throws InvalidArgumentException if text has invalid format.
+     */
+    static Arguments parse(String arglist) throws InvalidArgumentException {
+        ArgumentsBuilder builder = builder();
+
+        if (arglist != null && !arglist.isEmpty()) {
+            int beginProbe = 0;
+            int validBegin = 0;
+            while (true) {
+                int separatorIndex = arglist.indexOf(',', beginProbe);
+                if (separatorIndex == -1) {
+                    break;
+                }
+                String argumentText = arglist.substring(validBegin, separatorIndex);
+
+                ArgumentParser argumentParser = new ArgumentParser(argumentText, arglist);
+                if (argumentParser.isValid()) {
+                    Argument argument = argumentParser.parse();
+                    builder.addArgument(argument);
+                    validBegin = separatorIndex + 1;
+                }
+                beginProbe = separatorIndex + 1;
+            }
+
+            String argumentText = arglist.substring(beginProbe);
+            ArgumentParser argumentParser = new ArgumentParser(argumentText, arglist);
+            Argument argument = argumentParser.parse();
+            builder.addArgument(argument);
+        }
+
+        // Use builder capability to validate arguments uniqueness.
+        return builder.build();
+    }
+
+    private static class ArgumentParser {
+        private final String digitPart;
+        private final String valueDecorated;
+        private final String content;
+
+        public ArgumentParser(String argument, String content) {
+            this.content = content;
+            String[] keyValue = argument.split("=", 2);
+            digitPart = keyValue[0];
+            valueDecorated = keyValue.length == 1 ? null : keyValue[1];
+        }
+
+        private Argument parse() throws InvalidArgumentException {
+            if (!isValid()) {
+                throw new InvalidArgumentException("Unable to parse Arguments [%s]", content);
+            }
+            int digit;
+            try {
+                digit = Integer.parseInt(digitPart);
+            } catch (NumberFormatException e) {
+                throw new InvalidArgumentException(e, "Unable to parse Arguments [%s] with digit [%s]", content, digitPart);
+            }
+            String value = null;
+            if (valueDecorated != null) {
+                value = StringUtils.removeEnd(StringUtils.removeStart(valueDecorated, "'"), "'");
+            }
+
+            return new Argument(digit, value);
+        }
+
+        private boolean isValid() {
+            if (digitPart.length() != 1) {
+                return false;
+            }
```

Comment:
Not sure to get the benefits of this test ?

Reply:
You right. It was used in intermediate-developed code, but for now there is no sense to use it.